test/e2e/suites/apivalidations/openstackmachine_test.go

```diff
@@ -89,4 +95,187 @@ var _ = Describe("OpenStackMachine API validations", func() {
 		}
 		Expect(k8sClient.Create(ctx, machine)).To(Succeed(), "Creating a machine with max metadata key and value should succeed")
 	})
+
+	Context("volumes", func() {
+		It("should not allow volume with zero size", func() {
+			machine := defaultMachine()
+			machine.Spec.RootVolume = &infrav1.RootVolume{
+				SizeGiB: 0,
+			}
+			Expect(k8sClient.Create(ctx, machine)).NotTo(Succeed(), "Creating a machine with a zero size root volume should fail")
+
+			machine = defaultMachine()
+			machine.Spec.AdditionalBlockDevices = []infrav1.AdditionalBlockDevice{
+				{
+					Name:    "test-volume",
+					SizeGiB: 0,
+				},
+			}
+			Expect(k8sClient.Create(ctx, machine)).NotTo(Succeed(), "Creating a machine with a zero size additional block device should fail")
+		})
+
+		/* FIXME: These tests are failing
+		It("should not allow additional volume with empty name", func() {
+			machine.Spec.AdditionalBlockDevices = []infrav1.AdditionalBlockDevice{
+				{
+					Name:    "",
+					SizeGiB: 1,
+				},
+			}
+			Expect(k8sClient.Create(ctx, machine)).NotTo(Succeed(), "Creating a machine with an empty name additional block device should fail")
+		})
+
+		It("should not allow additional volume with name root", func() {
+			machine.Spec.AdditionalBlockDevices = []infrav1.AdditionalBlockDevice{
+				{
+					Name:    "root",
+					SizeGiB: 1,
+				},
+			}
+			Expect(k8sClient.Create(ctx, machine)).NotTo(Succeed(), "Creating a machine with a root named additional block device should fail")
+		})
+		*/
```

Comment:
We keep this on `FIXME`?

Reply:
I think so, yeah. Should probably create an issue to track it with a pointer to the FIXME code, though.

Reply:
Done: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/2013